### ISSUE TITLE
Remote Code Execution (Security)

### ISSUE THREAD

**Issue by bendecoste** [OpenMined/Grid#66]
For full pytorch support we will need remote code execution of arbitrary code (defined by someone who has defined a model)

We will also likely need remote code execution for data adapters, although these are more auditable (would still be nice to have, though)

some things I have considered so far (as well as their downsides)

#### chroot jail 
[what it is](https://unix.stackexchange.com/questions/105/chroot-jail-what-is-it-and-how-do-i-use-it) 
Basically we tell a process that `/grid/runhere` is the root directory and we manually link everything they are allowed to do (basically nothing) this is not that secure because there are ways to break out of it.

#### docker
We can look into docker, I have read docker is also no truly secure

**Comment by bendecoste** [comment]
related #48 & #58

@Mike96Angelo has experience with remote code execution

**Comment by bendecoste** [comment]
> docker

I would also like to stay clean of docker unless necessary.  Installation is way easier right now than if we were using docker imo

**Comment by iamtrask** [comment]
We could export tensors to a CSV https://stackoverflow.com/questions/36158058/torch-save-tensor-to-csv-file

**Comment by iamtrask** [comment]
and then we support specific layer wrappers - aka - "Sequential" and "Recurrent"

**Comment by jvmancuso** [comment]
How do we recreate the whole Module from that though?

**Comment by jvmancuso** [comment]
Marking this resolved, as the new approach for integrating pytorch allows us to make sure commands we receive are in the local `torch` module before executing it.  This will be actualized with #129, so feel free to direct further discussion there.
